tokio/src/io/seek.rs

```diff
@@ -33,11 +33,13 @@ where
         match me.pos {
             Some(pos) => {
                 match Pin::new(&mut me.seek).start_seek(cx, pos) {
-                    Poll::Ready(Ok(())) => me.pos = None,
-                    Poll::Ready(Err(e)) => return Poll::Ready(Err(e)),
-                    Poll::Pending => (),
-                };
-                Poll::Pending
+                    Poll::Ready(Ok(())) => {
+                        me.pos = None;
+                        Pin::new(&mut me.seek).poll_complete(cx)
+                    }
+                    Poll::Ready(Err(e)) => Poll::Ready(Err(e)),
+                    Poll::Pending => Poll::Pending,
```

Comment:
nit, take it or leave it: these lines _could_ be collapsed into just
```rust
poll => poll,
```
or similar. Not a blocker... perhaps the current code is clearer.

Reply:
This one actually can't be, because the left `poll` is `Result<(), _>` and the right side `poll` must be `Result<u64, _>`. Something like `poll => poll.map(|r| Err(r.unwrap_err())),` *works* but is more confusing IMO